### PR TITLE
AP-3476: Extend merits task list questions

### PR DESCRIPTION
**Pull request by colinbruce**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3476)

This adds a new endpoint, named to avoid generic use/clashes with crime if they start using it, that returns an identical task list as the previous endpoint, but allows rules to be set for a proceeding depending on the Client Involvement Type and use of delegated functions

It also uses a new pattern for Swagger tests that allows us to have a single test that exercises the request(RSpec) tests as well as allowing swagger generation

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: db/populators/proceeding_type_merits_task_populator.rb [hunk 1]
```diff
@@ -1,24 +1,32 @@
 class ProceedingTypeMeritsTaskPopulator
   DATA_FILE = Rails.root.join("db/seed_data/proceeding_type_merits_task.yml").freeze
 
+  attr_accessor :loop_index
```
Comment (kmahern) [inline db/populators/proceeding_type_merits_task_populator.rb:4]: do we need this attr_accessor?

File: spec/services/questions_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+          },
+          {
+            ccms_code: "SE013",
+            delegated_functions_used: true,
+            client_involvement_type: "D",
+          },
+        ]
+      end
+
+      it "returns valid response with expected tasks" do
```
Comment (kmahern) [inline spec/services/questions_service_spec.rb:56]: Can we make this something similar to "it returns valid response without questions 7 and 8" or "it returns valid response without domestic abuse/non applicant questions" and remove comment above?

File: spec/requests/civil_merits_questions_swagger_spec.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          description "Return questions base on proceeding codes, client involvement type and delegated function use"
+          let(:example_data) do
+            {
+              ccms_code: "DA001",
+              delegated_functions_used: true,
+              client_involvement_type: "X",
+            }
+          end
+
+          # TODO: @colinbruce 10 Oct 2022 review on: 11 Oct 2022
```
Comment (kmahern) [inline spec/requests/civil_merits_questions_swagger_spec.rb:66]: do we still want this TODO?

**Review by kmahern — COMMENTED** [review]
few minor suggestions otherwise looks good

File: spec/requests/civil_merits_questions_swagger_spec.rb (hunk 3 above)
Comment (colinbruce) [reply spec/requests/civil_merits_questions_swagger_spec.rb]: No! It's something I was considering but I have removed it for now :+1:

File: db/populators/proceeding_type_merits_task_populator.rb (hunk 1 above)
Comment (colinbruce) [reply db/populators/proceeding_type_merits_task_populator.rb:4]: No, another good catch 👏

**Review by kmahern — APPROVED** [review]
lgtm 👍

**Pull request merged**